File: python/nt_toolbox/load_sound.py
import numpy as np
import wave as wv


def load_sound(file, n0):

    x_raw = wv.open(file)
    n = x_raw.getnframes()
    x = np.frombuffer(x_raw.readframes(n), dtype="<i2").astype(float)
    x_raw.close()

    if file[::-1][:8][::-1] == "bird.wav":
        x = np.delete(
            x,
            list(range(6001))
            + list(range(12500, 15001))
            + list(range(22500, 24001))
            + list(range(32500, 34001)),
        )

    if n0 != 0 and n0 < n:
        x = x[:n0]

    return x / max(np.max(np.abs(x)), np.finfo(float).eps)

File: r/nt_toolbox/toolbox_signal/load_sound.R
library(tuneR)

load_sound <- function(file, n0){
  
  x_raw <- readWave(file)@left
  
  n <- length(x_raw)
  
  if (file == "nt_toolbox/data/bird.wav"){
   x_raw <- c(x_raw[6002:12499], x_raw[15001:22499], x_raw[24001:34000]) 
  }
   
  if((n0 < n) & (n0 !=0)){
    x_raw <- x_raw[1:n0]
  }
  return (x_raw/max(x_raw))
}
